app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java

```diff
@@ -0,0 +1,264 @@
+import android.support.v7.app.ActionBar;
+import android.preference.PreferenceFragment;
+import android.preference.PreferenceManager;
+import android.view.MenuItem;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Profile;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A {@link PreferenceActivity} that presents a set of application settings. This
+ * is the place where your profile is changed.
+ */
+@SuppressWarnings({"checkstyle:AnonInnerLength", "PMD.AvoidUsingVolatile"})
+//1. In this class large anonymous classes are needed. It does not infer with readability.
+//2. The volatile identifier is needed because this class uses static methods, which are essential.
+public final class SettingsActivity extends AAppCompatPreferenceActivity {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("SettingsActivity");
+    private static boolean firstLoad = true;
+    private static final String DISPLAY_NAME = "display_name";
+    private static volatile IServiceLocator serviceLocator;
+
+    /**
+     * A preference value change listener that updates the preference's summary
+     * to reflect its new value.
+     */
+    private static Preference.OnPreferenceChangeListener sBindPreferenceSummaryToValueListener = new Preference.OnPreferenceChangeListener() {
+        @Override
+        public boolean onPreferenceChange(final Preference preference, final Object value) {
+            LOGGER.info("Preference changed");
+            final String stringValue = value.toString();
+
+            if (preference.getKey().equals(DISPLAY_NAME)) {
+                LOGGER.info("Preference changed at the display name");
+                updateDisplayName(preference, stringValue);
+            } else if (preference instanceof ListPreference) {
+                LOGGER.info("Preference changed for a ListPreference");
+                // For list preferences, look up the correct display value in
+                // the preference's 'entries' list.
+                final ListPreference listPreference = (ListPreference) preference;
+                final int index = listPreference.findIndexOfValue(stringValue);
+
+                // Set the summary to reflect the new value.
+                preference.setSummary(
+                        index >= 0
+                                ? listPreference.getEntries()[index]
+                                : "");
+
+
+            } else {
+                LOGGER.info("Preference changed which is not a ListPreference, and not the display name");
+                // For all other preferences, set the summary to the value's
+                // simple string representation.
+                preference.setSummary(stringValue);
+            }
+            return true;
```

Comment:
Because of extension I have to return a boolean, but I'm not sure if that should be true or false. In the example I used they also returned true after setting the summaries of the elements.